site/app/libraries/database/DatabaseQueriesPostgresql.php

```diff
@@ -702,4 +734,62 @@ public function getAllGradeablesIds() {
         $this->database->query("SELECT g_id FROM gradeable ORDER BY g_id");
         return $this->database->rows();
     }
+
+    public function newTeam($g_id, $user_id) {
+        $this->database->query("SELECT * FROM gradeable_teams ORDER BY team_id");
+        $team_id_prefix = count($this->database->rows());
+        $team_id = "{$team_id_prefix}_{$user_id}";
+        $this->database->query("INSERT INTO gradeable_teams (team_id, g_id) VALUES(?,?)", array($team_id, $g_id));
+        $this->database->query("INSERT INTO teams (team_id, user_id, state) VALUES(?,?,1)", array($team_id, $user_id));
+    }
+
+    public function newTeamInvite($team_id, $user_id) {
+        $this->database->query("INSERT INTO teams (team_id, user_id, state) VALUES(?,?,0)", array($team_id, $user_id,));
+    }
+
+    public function newTeamMember($team_id, $user_id) {
+        $this->database->query("INSERT INTO teams (team_id, user_id, state) VALUES(?,?,1)", array($team_id, $user_id,));
+    }
+
+    public function removeTeamUser($g_id, $user_id) {
+        $team_ids = array();
+        $this->database->query("SELECT * FROM gradeable_teams WHERE g_id=? ORDER BY team_id", array($g_id));
+        foreach($this->database->rows() as $row) {
+            $team_ids[] = $row['team_id'];
+        }
+        foreach($team_ids as $team_id) {
+            $this->database->query("DELETE FROM teams WHERE team_id=? AND user_id=?", array($team_id, $user_id));
+        }
+    }
+
+    public function getTeamsByGradeableId($g_id) {
+        $teams = array();
+        $this->database->query("SELECT * FROM gradeable_teams WHERE g_id=? ORDER BY team_id", array($g_id));
+        foreach($this->database->rows() as $row) {
+            $teams[] = new Team($row['team_id']);
+        }
+        for($i = 0; $i < count($teams); $i++) {
+            $this->database->query("SELECT * FROM teams WHERE team_id=? ORDER BY user_id", array($teams[$i]->getId()));
+            $teams[$i]->addUsers($this->database->rows());
+        }
+        return $teams;
+    }
+
+    public function getTeamByUserId($g_id, $user_id) {
+        $team_ids = array();
+        $this->database->query("SELECT * FROM gradeable_teams WHERE g_id=? ORDER BY team_id", array($g_id));
+        foreach($this->database->rows() as $row) {
+            $team_ids[] = $row['team_id'];
+        }
+        for($i = 0; $i < count($team_ids); $i++) {
+            $this->database->query("SELECT * FROM teams WHERE team_id=? AND user_id=? AND state=? ORDER BY user_id", array($team_ids[$i], $user_id, 1));
+            if (count($this->database->rows()) === 1) {
```

Comment:
I'm not sure what you're trying to do here. So you get a list of all teams for a greadeable above, and then use that to try and find the teams that contain this user?

We should be able to replace this logic with less loops/queries using joins and IN, but will need you to tell me exactly what you're trying to do here.